types/webpack/index.d.ts

```diff
@@ -134,7 +134,7 @@ declare namespace webpack {
     }
 
     interface Output {
-        /** The output directory as absolute path (required). */
```

Comment:
Is this required in v4 still? I know it is in v3..

Could you add these changes to the v3 subfolder also. Thanks!

(Sorry if youre getting double notification! I accidentally cancelled the review..)

Reply:
Are there versioned versions of the webpack docs? I hadn't touched the v3 file because I wasn't sure whether these options existed at the time.

I've just tested my config on both webpack v4 and v3.11.0, and I am able to build my project without problem. This demonstrates the existence of `hashFunction`, `hashDigestLength`, and `hashSalt`, as well as the ability to omit `path`. At least in the two versions I tested, I did not need to specify the `path` option. `filename`, on the other hand, appears to be required for v3 (v4 names the file `main.js` by default).

I note that the default `path` in v3 appears to be the root of the project, whereas the default `path` in v4 is `dist/`.

I will change the comment and add the new options to the v3 type definition after lunch.

Reply:
I think the webpack v4 docs are still not ready.. most of the docs you see on their site are all for v3 unfortunately. I have been going straight to the source to look up what/how things are defined or configured and I think you're right on in what you've discovered.

Like you said, filename is what's required in v3, not path and its no longer required in v4. I think it is defined as optional here in `filename?` because users might need partial configurations. Adding the comment in v3 wouldnt be a bad thing though.

Reply:
Thanks for the heads up - had not realized the docs were not the most up-to-date.

I've added a comment (made it a requirement before re-reading your comment, then reverted. Sorry for the messy commit history).

I believe we are good to go, but please let me know if I've missed anything.